### PR TITLE
feat:  add suport for node16 runtime

### LINKED ISSUE #631: Add suport for Node.js 16
On March 30, 2022, Microsoft announced that Node.js 16 was available for Azure Functions [https://azure.microsoft.com/en-us/updates/generally-available-nodejs-16-in-azure-functions-2/](https://azure.microsoft.com/en-us/updates/generally-available-nodejs-16-in-azure-functions-2/), but this plugin does not support this version. I found an open pull request about this issue, but with a small mistake in the code [https://github.com/serverless/serverless-azure-functions/pull/610](https://github.com/serverless/serverless-azure-functions/pull/610). Could you please add support for Node.js 16?

### PR DESCRIPTION
**Pull request by lcsales98**
## What did you implement:

Closes #631 

It will become available to use the nodejs16 runtime.

## How did you implement it:

I'm updated the available runtimes at [src/config/runtime.ts](https://github.com/serverless/serverless-azure-functions/pull/640/commits/a8a7d68ff874ed53716c6901c4e734a471a4aa18).

## How can we verify it:

Can you generate an azure function with serverless-framework and deploy it with nodejs16 runtime.
Ex: `runtime: nodejs16`

## Todos:

_**Note: Run `npm run test:ci` to run all validation checks on proposed changes**_

- [x] Ensure there are no lint errors.  
       **Validate via `npm run lint`**  
       _Note: Some reported issues can be automatically fixed by running `npm run lint:fix`_
- [x] Write tests and confirm existing functionality is not broken.  
       **Validate via `npm test`**
- [x] Write documentation
- [x] Provide verification config / commands / resources
- [x] Enable "Allow edits from maintainers" for this PR
- [x] Update the messages below

**_Is this ready for review?:_** YES  
**_Is it a breaking change?:_** NO

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,11 +1,11 @@
 {
   "name": "serverless-azure-functions",
-  "version": "2.1.3",
```
Comment (medikoo) [inline package-lock.json:3]: Let's not introduce changes in this file

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "serverless-azure-functions",
-  "version": "2.1.3",
```
Comment (medikoo) [inline package.json:3]: Let's not bump version (releases are handled separately)

File: src/armTemplates/resources/functionApp.test.ts [hunk 3]
```diff
@@ -80,7 +80,7 @@ describe("Function App Resource", () => {
         const config = getConfig(FunctionAppOS.LINUX, Runtime.PYTHON36);
 
         const resource = new FunctionAppResource();
-
```
Comment (medikoo) [inline src/armTemplates/resources/functionApp.test.ts:83]: Let's not introduce whitespace and unrelated changes

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @lcsales98! Please see my comments

File: src/armTemplates/resources/functionApp.test.ts (hunk 3 above)
Comment (lcsales98) [reply src/armTemplates/resources/functionApp.test.ts:83]: unrelated changes like the tests `it("throw exception on get runtime version")` and `it("throw exception on get runtime language")`

Comment (medikoo) [reply src/armTemplates/resources/functionApp.test.ts]: @lcsales98 see line under which I've put comment. There should be no change here

File: package-lock.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
   "requires": true,
   "packages": {
     "": {
-      "version": "2.1.0",
```
Comment (medikoo) [inline package-lock.json:8]: Let's not commit changes into this file

**Review by medikoo — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 4 above)
Comment (lcsales98) [reply package-lock.json]: Sorry, I didn't notice that the file had a lot of changes

**Comment by danielwaghorn** [comment]
Are there any updates on being able to get this merged at all? Happy to help if required.

**Comment by medikoo** [comment]
> Are there any updates on being able to get this merged at all? Happy to help if required.

The only problem was that author of PR didn't mark this PR as ready for re-review. Anyway it looks as good to take

**Comment by medikoo** [comment]
I've merged it, yet it doesn't seem to be validated in CI, and locally for me tests fails.

Before releasing it, it'll be  good to fix the CI in this project (I've created an issue: https://github.com/serverless/serverless-azure-functions/issues/645) and ensure that locally all tests passes
